src/context/AppProvider/AppContext.tsx

```diff
@@ -300,34 +304,33 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   // for more obscure assets, if we don't have it, fetch it
   const portfolioAssetIds = useSelector(selectPortfolioAssetIds)
 
-  // creating a variable to store the intervals in
-  const [marketDataIntervalId, setMarketDataIntervalId] = useState<NodeJS.Timer | undefined>()
-
   // market data pre and refetch management
   useEffect(() => {
-    if (!portfolioAssetIds.length) return
-
-    const fetchMarketData = () => {
+    const fetchMarketData = () =>
       portfolioAssetIds.forEach(assetId => {
-        dispatch(marketApi.endpoints.findByAssetId.initiate(assetId, { forceRefetch: true }))
+        dispatch(marketApi.endpoints.findByAssetId.initiate(assetId))
+        const timeframe = DEFAULT_HISTORY_TIMEFRAME
+        const payload = { assetId, timeframe }
+        dispatch(marketApi.endpoints.findPriceHistoryByAssetId.initiate(payload))
       })
-    }
-
-    // do this the first time once
-    fetchMarketData()
 
-    // clear the old timer
-    if (marketDataIntervalId) {
-      clearInterval(marketDataIntervalId)
-      setMarketDataIntervalId(undefined)
-    }
+    fetchMarketData() // fetch every time assetIds change
+    // refetch every two minutes
+    return () => clearInterval(setInterval(fetchMarketData, 1000 * 60 * 2))
+  }, [portfolioAssetIds, dispatch])
```

Comment:
simplified this existing logic, it was a mess, and does the same thing but doesn't overfetch now

Reply:
Will this correctly clear the interval since its not in state anymore?

its clearing the new interval?

Reply:
as discussed this is correct

Reply:
Tested it out, this is still working as the previous implementation

Reply:
Actually I stand corrected, the initial fetch is still working as intended, but I tried testing against develop many times, and the subsequent, 2mn interval isn't working anymore it seems (tried logging in `fetchMarketData` body)

Reply:
derp, of course, the intervel was not being set until the cleanup function ran, fixed.